Sources/Router/OARoutingHelper.mm

```diff
@@ -279,6 +279,7 @@ - (void) setAppMode:(OAApplicationMode *)mode
     ARRIVAL_DISTANCE_FACTOR = MAX([_settings.arrivalDistanceFactor get:mode], 0.1f);
     GPS_TOLERANCE = (NSInteger) (DEFAULT_GPS_TOLERANCE * ARRIVAL_DISTANCE_FACTOR);
     [_voiceRouter updateAppMode];
+    [_app.data.applicationModeChangedObservable notifyEventWithKey:mode];
```

Comment:
This is incorrect, applicationModeChangedObservable should only be used to notify the change of the application mode, not routing helper mode

Reply:
You do not need the observers here at all, set the mode for the routing helper and reload the ui